lib/Dialect/FIRRTL/Import/FIRParserAsserts.cpp

```diff
@@ -134,10 +153,9 @@ struct ExtractionSummaryCursor {
 };
 
 /// Convenience function to create a `ExtractionSummaryCursor`.
-template <typename JsonType>
-ExtractionSummaryCursor<JsonType>
-makeExtractionSummaryCursor(Location loc, JsonType jsonValue) {
-  return ExtractionSummaryCursor<JsonType>{loc, {}, jsonValue};
+template <typename JsonType, typename FnType>
+ParseResult parseJson(Location loc, const JsonType &jsonValue, FnType fn) {
+  return fn(ExtractionSummaryCursor<JsonType>{loc, {}, jsonValue});
 }
```

Comment:
I like this lambda continuation style pattern where we have precise control over the lifetime of the parameters passed into the lambda. Very Rust-y :crab: :grin: